dolosse/run_control/cli/kafka_pyinq.py

```diff
@@ -0,0 +1,263 @@
+            with open(metadata_filename) as mf:
+                metadata_json = dumps(loads(mf.read()))
+
+            # compose control message
+            control_json = dumps({"category": "control", "run": {"action": answers['command'],
+                                                                 "run_number": run_number}})
+            control_msg = Message(yaml_conf['topic_control'], control_json)
+            control_msg.execute()
+
+            manage_msg = Message(yaml_conf['topic_manage'], control_json)
+            manage_msg.execute()
+
+            manage_msg.json = equipment_json
+            manage_msg.execute()
+
+            manage_msg.json = metadata_json
+            manage_msg.execute()
+
+        elif answers['command'] == 'Exit':
+            # Wait until all messages have been delivered
+            sys.stderr.write('%% Waiting for %d deliveries\n' % len(producer))
+            producer.flush()
+            print('That\'s all folks!')
+            run = False
+
+        elif answers['command'] == 'Refresh':
+            clear()
+
+        else:
+            control_json = dumps({"category": "control", "run": {"action": answers['command'],
+                                                                 "run_number": run_number}})
+            control_msg = Message(yaml_conf['topic_control'], control_json)
+            control_msg.execute()
+
+
+def status_readout(status):
+    while run:
+        msg = consumer.poll(timeout=1.0)
+        if msg is None:
+            continue
+        if msg.error():
+            raise KafkaException(msg.error())
+        else:
+            # Proper message
+            sys.stderr.write('%% %s [%d] at offset %d with key %s:\n' %
+                             (msg.topic(), msg.partition(), msg.offset(), str(msg.key())))
+            json_data = loads(msg.value())
+            if json_data['category'] == 'feedback':
+                status['daq_state'] = json_data['daq']['run_state']
+                status['daq_feedback'] = json_data['os']
+                status['daq_evt_rate'] = str(json_data['daq']['evt_rate'])
+                status['daq_kB_rate'] = str(json_data['daq']['kB_rate'])
+                status['daq_events'] = str(json_data['daq']['events'])
+            elif json_data['category'] == 'errors':
+                status['daq_error'] = json_data['msg']
+
+    consumer.close()
+
+
+def read_yaml_config(conf_dict):
+    # read yaml file
+    with open('kafka_interface.yaml') as yf:
+        config = safe_load(yf)
+    conf_dict['bootstrap_servers'] = config['kafka']['bootstrap_servers']
+    conf_dict['topic_control'] = config['kafka']['topics']['control']
+    conf_dict['topics'] = [config['kafka']['topics']['feedback'], config['kafka']['topics']['errors']]
+    conf_dict['group_id'] = config['kafka']['group_id_interface']
+    conf_dict['topic_manage'] = config['kafka']['topics']['manage']
+
+
+if __name__ == '__main__':

```

Comment:
Recommend adding CLI arguments for location of configuration file. Ideally we want people to be able to run as `python3 -m dolosse.acquisition.midas_run_control /path/to/config.yaml`

Reply:
Done